### ISSUE TITLE
RootCauses Artifact is slow if there are lots of "SVG changed" invalidations

### ISSUE THREAD

**Issue by mattzeunert** [GoogleChrome/lighthouse#15869]
### FAQ

- [X] Yes, my issue is not about [variability](https://github.com/GoogleChrome/lighthouse/blob/main/docs/variability.md) or [throttling](https://github.com/GoogleChrome/lighthouse/blob/main/docs/throttling.md).
- [X] Yes, my issue is not about a specific accessibility audit (file with [axe-core](https://github.com/dequelabs/axe-core) instead).

### URL

https://lottiefiles.com/

### What happened?

Run Lighthouse `lighthouse https://lottiefiles.com/`

The root causes gatherer takes 34 seconds

```
 LH:status Getting artifact: DevtoolsLog +7ms
  LH:status Getting artifact: Trace +0ms
  LH:status Getting artifact: RootCauses +0ms
  LH:status Getting artifact: Accessibility +34s
  LH:status Getting artifact: AnchorElements +670ms
  LH:status Getting artifact: ConsoleMessages +138ms
```

Counting the invalidations/prepaints:

```
{ eventsForLayoutInvalidation: 784184, prePaintEvents: 515 }
```

### What did you expect?

I expected it to be faster.

### What have you tried?

Some initial profiling on another lottie player site:

![CleanShot 2024-03-20 at 12 23 39](https://github.com/GoogleChrome/lighthouse/assets/1303660/d4ca63bf-f845-47f1-bb85-db402cd6c6dd)
![CleanShot 2024-03-20 at 11 50 48](https://github.com/GoogleChrome/lighthouse/assets/1303660/81936e5e-9fb3-45e8-aa4b-5553d8ada986)
![CleanShot 2024-03-20 at 11 52 08](https://github.com/GoogleChrome/lighthouse/assets/1303660/37db9e6a-2f74-4fef-be52-f46523628900)

### How were you running Lighthouse?

CLI

### Lighthouse Version

11.6.0

**Comment by adamraine** [comment]
Definitely reproducible, looks like we are truing to get the node details of every layout invalidation node which is taking a while.

**Comment by adamraine** [comment]
For me Lighthouse takes ~4 minutes locally on https://lottiefiles.com/
Some optimizations in the root causes engine helped get that down to ~2:45 (~2 min of which is spent in root causes)

We should continue to iterate on the performance of the root causes engine, but https://lottiefiles.com/ also contains *a lot* of invalidation tracking events (specifically iframe added to dom events) that the root causes engine has to sift through. We way want to limit the root causes engine somehow since it's results are helpful but not vital.

**Comment by adamraine** [comment]
The performance should of `RootCauses` should be much improved with https://github.com/GoogleChrome/lighthouse/pull/15926 released in LH 11.7.1.

It might be possible to improve performance further, but I expected us to overhaul the root causes engine in the near future so I think we can close this.
